modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/InboundConnectionHandler.java

```diff
@@ -711,25 +711,19 @@ else if (discoverySpi instanceof IgniteDiscoverySpi)
                     ses.send(new RecoveryLastReceivedMessage(ALREADY_CONNECTED));
                 }
                 else {
-                    try {
```

Comment:
Why did you remove the try-finally block?

Reply:
Because it was incorrect to always remove `oldFut` from the map. We must only remove it if we managed to create a connection.

Reply:
ok, so it was originally there just to save us from code duplication?

Reply:
I think it was there *due* to code duplication, as a result of a copy-paste